src/Inline/Parser/MentionGenerator/CallbackGenerator.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the league/commonmark package.
+ *
+ * (c) Colin O'Dell <colinodell@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace League\CommonMark\Inline\Parser\MentionGenerator;
+
+use League\CommonMark\Inline\Element\AbstractInline;
+use League\CommonMark\Inline\Element\Mention;
+
+final class CallbackGenerator implements MentionGeneratorInterface
+{
+    /**
+     * A callback function which returns the URL to use, or null if no link should be generated
+     *
+     * @var callable(Mention): ?(AbstractInline|false)
+     */
+    private $callback;
+
+    public function __construct(callable $callback)
+    {
+        $this->callback = $callback;
+    }
+
+    public function generateMention(Mention $mention): ?AbstractInline
```

Comment:
I think I'd prefer the original method signature of taking in the parsed symbol and identifier and letting this class decide how to create the `Mention` (or possibly some other `AbstractInline`)

Reply:
I really don't like the idea of passing variables by reference. I really feel that this should pass a `Mention` node as it will likely be what is eventually render. It's up to the generator to supply the mention with a URL or return a different `AbstractInline`. This also allows data to be added to the mention prior to rendering, i.e. what github does for it's mention tooltips:

```
<a class="issue-link js-issue-link" data-error-text="Failed to load title" data-id="126995301" data-permission-text="Title is private" data-hovercard-type="pull_request" data-hovercard-url="/thephpleague/commonmark/pull/222/hovercard" href="https://github.com/thephpleague/commonmark/pull/222" aria-label="#222, Resolve parent class">#222</a>
```